python/oneseismic/scan/tests/test_scan.py

```diff
@@ -40,3 +42,65 @@ def test_regular_intervals(inlines, crosslines):
         seg.scan_trace_header(header)
 
     assert len(seg.key2s) == crosslines
+
+@given(
+    integers(min_value = 2, max_value = 15),
+    integers(min_value = 2, max_value = 15),
+    floats(min_value = 1, max_value = 10),
+    floats(min_value = 1, max_value = 10),
+    floats(min_value = -math.pi, max_value = math.pi),
+    floats(min_value = -10, max_value = 10),
+    floats(min_value = -10, max_value = 10)
+)
+def test_rot_step(
+        inlines,
+        crosslines,
+        ilstep,
+        xlstep,
+        rotation,
+        offsetx,
+        offsety
+):
+    ilstepx = ilstep * math.cos(rotation)
+    ilstepy = ilstep * math.sin(rotation)
+    xlstepx = xlstep * math.cos(rotation)
+    xlstepy = xlstep * math.sin(rotation)
+
+    headers = [
+        {
+            int(segyio.su.iline): big_endian(i),
+            int(segyio.su.xline): big_endian(x),
+            int(segyio.su.cdpx): i * ilstepx + x * xlstepx + offsetx,
+            int(segyio.su.cdpy): i * ilstepy + x * xlstepy + offsety,
+            int(segyio.su.ns): 0,
+            int(segyio.su.dt): 0,
+            int(segyio.su.scalco): 1,
+        }
+        for i in range(1, inlines + 1)
+        for x in range(1, crosslines + 1)
+    ]
+
+    seg = GeoScanner(
+        il_word=int(segyio.su.iline),
+        xl_word=int(segyio.su.xline),
+        endian=sys.byteorder,
```

Comment:
Because endianness is accounted for when extracting the values from the headers. In this case it is not needed because the headers are not read from disk, but the function still requires a value.